dashboard/src/components/AppUpgrade/AppUpgrade.test.tsx

```diff
@@ -75,12 +76,40 @@ it("renders an error if it exists", () => {
       repo={repo}
     />,
   );
-  expect(wrapper.find(DeploymentErrors).exists()).toBe(true);
+  expect(wrapper.find(ErrorSelector).exists()).toBe(true);
   expect(wrapper.find(SelectRepoForm).exists()).toBe(false);
   expect(wrapper.find(UpgradeForm).exists()).toBe(false);
+  expect(wrapper.html()).toContain("Sorry! Something went wrong.");
   expect(wrapper).toMatchSnapshot();
 });
 
+it("renders a forbidden error", () => {
+  const repo = {
+    metadata: { name: "stable" },
+  } as IAppRepository;
+  const wrapper = shallow(
+    <AppUpgrade {...defaultProps} error={new ForbiddenError()} repos={[repo]} repo={repo} />,
+  );
+  expect(wrapper.find(ErrorSelector).exists()).toBe(true);
+  expect(wrapper.find(SelectRepoForm).exists()).toBe(false);
+  expect(wrapper.find(UpgradeForm).exists()).toBe(false);
+  expect(wrapper.html()).toContain(
+    "You don&#x27;t have sufficient permissions to update foo in <span>the <code>default</code> namespace</span>",
+  );
+  expect(
+    wrapper
+      .find(ErrorSelector)
+      .shallow()
+      .find(PermissionsErrorPage)
```

Comment:
Can't you find `PermissionsErrorPage` directly from the wrapper? Or that's why you call shallow() to go one level deep?